Documentation/html_viewer/js/dictionary_functions.js

```diff
@@ -483,7 +486,29 @@
 		
 		var tHTML = "";
 		var references = [];
-		tHTML += '<h1 style="margin:0px 0px 30px 12px">'+tEntryObject["display name"]+'</h1><div class="row">';
+		
+		tHTML += '<h1 style="margin:0px 0px 30px 12px">'+tEntryObject["display name"];
+		
+		var tUpgrade = true;
+		if (!tEntryObject.hasOwnProperty("edition"))
+			tUpgrade = false;
+		else if (tState.edition == "professional")
+			tUpgrade = false;
+		else if (tState.edition == "commercial" && tEntryObject["edition"].toLowerCase() != "business")
+			tUpgrade = false;
+		else if (tState.edition == "communityplus" && (tEntryObject["edition"].toLowerCase() != "business" || tEntryObject["edition"].toLowerCase() != "indy"))
+			tUpgrade = false;
+		else if (tEntryObject["edition"].toLowerCase() == "community")
+			tUpgrade = false;
+		
+		if (tUpgrade)
+		{
+			tHTML += ' <a href="#" class="btn btn-default" onclick="javascript:liveCode.showUpgradeOptions(\'' 
```

Comment:
Could you make `upgrade` a class and `edition` and `display_name` attributes of the element, and then add an 
```
 		$("body").on("click", ".upgrade", function() {
 			var tEdition = $(this).attr("edition");
                        ...
 		});
```
to the `setActions` function please? It makes it much easier to see what is happening, modify and add other places that need to show upgrade options if we want them

Reply:
Plus you can gate the call to `showUpgradeOptions` with `isRunningInLiveCodeBrowser()` so there's no error in an external browser

Reply:
@livecodeali sorry have had my head down for a few days and just read this. Yes to the first comment. Yes to the second but it makes me wonder if we could build the URL locally so the button can work when in an external browser.

Reply:
OK so I'm just redirecting them to livecode.com for now so it doesn't do nothing if not running in the browser widget.

Reply:
That's a good idea about making the button work in an external browser yeah...  could just redirect here: https://livecode.com/products/livecode-platform/pricing/